src/writing-policies/rust.md

```diff
@@ -1,25 +1,647 @@
-
 # Rust
 
 [Rust](https://www.rust-lang.org/) is the most mature programming language that
-can generate WebAssembly modules.
+can generate WebAssembly modules: WebAssembly is a first-class citizen
+in the Rust world. That means many of the tools and crates of the Rust
+ecosystem work out of the box.
+
+Kubewarden provides a [Rust SDK](https://crates.io/crates/kubewarden-policy-sdk)
+that simplifies the process of writing policies, plus a
+[template project](https://github.com/kubewarden/policy-rust-template) to
+quickly scaffold a policy project.
+
+This document illustrates how to take advantage of these projects to write
+Kubewarden policies using the Rust programming language.
+
+Note well, we won't cover the details of Kubewarden's Rust SDK inside of this
+page. These can be found inside of the
+[official crate documentation](https://docs.rs/kubewarden-policy-sdk/0.1.0).
+
+## Getting Rust dependencies
+
+This section guides you through the process of installing the Rust compiler and
+its dependencies.
+
+As a first step install the Rust compiler and its tools, this can be easily done
+using [rustup](https://github.com/rust-lang/rustup). Please follow
+[rustup's install documentation](https://rust-lang.github.io/rustup/installation/index.html).
+
+Once `rustup` is installed add the Wasm target:
+
+```shell
+rustup target add wasm32-unknown-unknown
+```
+ 
+Finally, install the [cargo-generate](https://github.com/cargo-generate/cargo-generate) utility:
+
+```shell
+cargo install cargo-generate
+```
+
+## Creating a new validation policy
+
+We are going to create a simple validation policy that processes
+Pod creation requests.
+
+The policy will look at the `metadata.name` attribute of the Pod and reject
+the pods that have an invalid name. We want the list of invalid names to be
+configurable by the end users of the policy.
+
+To summarize, the policy settings will look like that:
+
+```yaml
+invalid_names:
+- bad_name1
+- bad_name2
+```
+
+The policy will accept the creation of a Pod like the following one:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+    - name: nginx
+      image: nginx:latest
+```
+
+While it will reject the creation of a Pod like the following one:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: bad_name1
+spec:
+  containers:
+    - name: nginx
+      image: nginx:latest
+```
+
+### Scaffolding new policy project
+
+The creation of a new policy project can be done by feeding this
+[template project](https://github.com/kubewarden/policy-rust-template)
+into `cargo generate`:
+
+```shell
+cargo generate --git https://github.com/kubewarden/policy-rust-template \
+               --branch main \
+               --name demo
+```
+
+The command will produce the following output:
+
+```
+🔧   Creating project called `demo`...
+✨   Done! New project created /home/flavio/hacking/kubernetes/kubewarden/demo
+```
```

Comment:
It just got updated and it will ask a couple more questions with default answers... Should we update this too?

Reply:
No, I think the users will get it. We might expand the set of questions in the future, I don't want to go crazy and try to keep it always 1:1